indicator/src/main/java/com/fuzz/indicator/widget/TextClippedImageView.java

```diff
@@ -0,0 +1,146 @@
+     * background color for non-clipped areas that should appear transparent.
+     */
+    @ColorInt
+    public static final int NOT_CLIPPED = Color.BLACK;
+
+    /**
+     * Set this to some sort of bounded region in order to restrict background
+     * painting to that region. The foreground will then be masked by the background,
+     * ultimately meaning that all drawn content will fit within this path.
+     */
+    @Nullable
+    protected Path backgroundMaskPath;
+
+    /**
+     * A string whose bounds define {@link #backgroundMaskPath}.
+     */
+    @Nullable
+    protected String backgroundMaskText;
+
+    /**
+     * A paint object for drawing barely-visible text.
+     */
+    @NonNull
+    protected TextPaint textPaint = new TextPaint(Paint.ANTI_ALIAS_FLAG);
+
+    public TextClippedImageView(Context context) {
+        super(context);
+    }
+
+    public TextClippedImageView(Context context, AttributeSet attrs) {
+        super(context, attrs);
+    }
+
+    public TextClippedImageView(Context context, AttributeSet attrs, int defStyleAttr) {
+        super(context, attrs, defStyleAttr);
+    }
+
+    @SuppressWarnings("unused")
+    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
+    public TextClippedImageView(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
+        super(context, attrs, defStyleAttr, defStyleRes);
+    }
+
+    @NonNull
+    @Override
+    protected Bitmap clipImageInto(int viewWidth, int viewHeight) {
+
+        intermediary = ensureBitmapIsUsable(intermediary, viewWidth, viewHeight);
+        intermediary.eraseColor(Color.TRANSPARENT);
+
+        Canvas canvas = new Canvas(intermediary);
+
+        paint.setXfermode(null);
+        textPaint.setXfermode(null);
+
+        textPaint.setTextSize(30);
+        textPaint.setTypeface(Typeface.DEFAULT_BOLD);
+        textPaint.setFakeBoldText(true);
+        textPaint.setTextAlign(Paint.Align.CENTER);
+        textPaint.setColor(NOT_CLIPPED);
+
+        if (backgroundMaskPath != null) {
+            canvas.drawPath(backgroundMaskPath, textPaint);
+            paint.setXfermode(overlayTransferMode);
+        } else if (backgroundMaskText != null) {
+            canvas.drawText(backgroundMaskText, getWidth() / 2, getHeight() / 2, textPaint);
+            paint.setXfermode(overlayTransferMode);
+        }
+
+        canvas.drawBitmap(backgroundBitmap, 0, 0, paint);
+
+        paint.setXfermode(overlayTransferMode);
+        int count = canvas.save();
+        canvas.concat(getImageMatrix());
+        canvas.drawBitmap(primaryBitmap, 0, 0, paint);
+        canvas.restoreToCount(count);
+
+
+        return intermediary;
+    }
+
+    /**
+     * The passed string will be used to mask the background (and therefore the foreground as well).
+     * <br/><br/>
+     * Pass in null to disable this feature.
+     */
+    public void setTextMaskPath(@Nullable String mask) {
+        backgroundMaskText = mask;
+        if (mask != null && !isInEditMode()) {
```

Comment:
After looking at this a bit....there're too many assumptions to be made around whether the text takes priority over the path. That can be its own issue, later.